### PR TITLE
MULE-10306: Add option to disable internal entity expansion in XML (l…

### PR DESCRIPTION
**Pull request by alepulver**
…eads to DoS)

### CODE REVIEW

File: modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java [hunk 1]
```diff
@@ -76,6 +81,12 @@ public final void initialise() throws InitialisationException
             useStaxSource = true;
         }
 
+        String expandInternalEntitiesValue = System.getProperty(EXPAND_INTERNAL_ENTITIES_PROPERTY, "true");
+        if (expandInternalEntitiesValue.toLowerCase().equals("false")) {
```
Comment (afelisatti) [inline modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java]: Format.

File: modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java [hunk 2]
```diff
@@ -76,6 +81,12 @@ public final void initialise() throws InitialisationException
...
+            expandInternalEntities = false;
```
Comment (afelisatti) [inline modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java]: Shouldn't you just parse the value?

File: modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java [hunk 3]
```diff
@@ -76,6 +81,12 @@ public final void initialise() throws InitialisationException
...
+            xmlInputFactory.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, false);
```
Comment (afelisatti) [inline modules/xml/src/main/java/org/mule/module/xml/transformer/AbstractXmlTransformer.java]: Static import.

File: modules/xml/src/test/java/org/mule/transformers/xml/xslt/XsltTransformerBLDisabledTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @Rule
+    public final SystemProperty myPropertyHasMyValue = new SystemProperty(EXPAND_INTERNAL_ENTITIES_PROPERTY, "false");
+
+    @Test
+    public void disabled() throws Exception
+    {
+        String input = this.makeInput();
+        Object payload = input.getBytes();
+        String output = (String) this.runFlow("flow", payload).getMessage().getPayload();
+        assertFalse(output.contains("010101010101010101010101010101010101010101010101"));
```
Comment (afelisatti) [inline modules/xml/src/test/java/org/mule/transformers/xml/xslt/XsltTransformerBLDisabledTest.java]: Use hamcrest matchers. Same in the other test.

**Comment by aiannucci** [comment]
Test classes should finish with ***TestCase***

**Pull request merged**